plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java

```diff
@@ -881,6 +883,111 @@ public void testSystemSchemas()
+                                statsWithoutPushdown -> {
+                                    assertEquals(statsWithoutPushdown.getProcessedInputPositions(), 3);
+                                    assertEquals(processedInputPositionWithPushdown, 1);
+                                    assertThat(statsWithoutPushdown.getProcessedInputPositions()).isGreaterThan(processedInputPositionWithPushdown);
+                                },
+                                results -> assertEquals(results.getOnlyColumnAsSet(), expected));
+                    },
+                    results -> assertEquals(results.getOnlyColumnAsSet(), expected));
+
+            assertQueryStats(
+                    getSession(),
+                    format("SELECT 1 FROM %s WHERE col_0.col_2.col_4.col_5 = %s", table.getName(), expectedValue),
+                    statsWithPushdown -> {
+                        long processedInputPositionWithPushdown = statsWithPushdown.getProcessedInputPositions();
+                        assertQueryStats(
+                                sessionWithoutPushdown,
+                                format("SELECT 1 FROM %s WHERE col_0.col_2.col_4.col_5 = %s", table.getName(), expectedValue),
+                                statsWithoutPushdown -> {
+                                    assertEquals(statsWithoutPushdown.getProcessedInputPositions(), 3);
+                                    assertEquals(processedInputPositionWithPushdown, 2);
+                                    assertThat(statsWithoutPushdown.getProcessedInputPositions()).isGreaterThan(processedInputPositionWithPushdown);
+                                },
+                                results -> assertEquals(results.getOnlyColumnAsSet(), expected));
+                    },
+                    results -> assertEquals(results.getOnlyColumnAsSet(), expected));
+        }
+    }
+
+    @DataProvider
+    public Object[][] nestedValuesProvider()
+    {
+        return new Object[][] {
+                {"varchar 'String type'", "varchar"},
+                {"to_utf8('BinData')", "varbinary"},
+                {"bigint '1234567890'", "bigint"},
+                {"true", "boolean"},
+                {"double '12.3'", "double"},
+                {"timestamp '1970-01-01 00:00:00.000'", "timestamp(3)"},
+                {"array[bigint '1']", "array(bigint)"},
+                {"ObjectId('5126bc054aed4daf9e2ab772')", "ObjectId"},
+        };
+    }
+
+    @Override
+    @Test
+    public void testProjectionPushdownReadsLessData()
+    {
+        // TODO https://github.com/trinodb/trino/issues/17713
+        throw new SkipException("MongoDB connector does not calculate physical data input size");
+    }
+
+    @Override
+    @Test
+    public void testProjectionPushdownPhysicalInputSize()
+    {
+        // TODO https://github.com/trinodb/trino/issues/17713
+        throw new SkipException("MongoDB connector does not calculate physical data input size");
+    }
+
```

Comment:
Could you add a test case that a collection has different types in the same field name? We can prepare such data using `MongoClient`. 
e.g. the 1st document has int type and 2nd document has nested field on `x` column

Reply:
Hi @ebyhr, What would be the expected output? 

> the 1st document has integer type and 2nd document has row type on the same column

For this case, I am getting null for row type column

Is this the case you meant?
``` java
    @Test
    public void testProjectionPushdownWithDifferentType()
    {
        String tableName = "test_projection_pushdown_with_different_type_" + randomNameSuffix();
        MongoCollection<Document> collection = client.getDatabase("test").getCollection(tableName);
        collection.insertOne(new Document().append("col1", 100));
        collection.insertOne(new Document().append("col1", new Document().append("child", "value")));
        assertThat(query("SELECT col1 FROM test." + tableName))
                .skippingTypesCheck()
                .matches("VALUES (BIGINT '100'), (NULL)");
    }
```

Reply:
Getting null on primitive type is a known issue as TODO comment in MongoPageSource. 

I would like to know what happens if we specify the nested field like `SELECT col1.chil ...`?

Reply:
`query("SELECT col1.child FROM test." + tableName)` fails with below error (with or without my change).

```
io.trino.testing.QueryFailedException: line 1:8: Expression col1 is not of type ROW

```

Reply:
Please create the table with row type instead of relying on guessing column type definition.

Reply:
Added `testProjectionPushdownWithColumnMissingInDocument` and `testProjectionPushdownWithDifferentTypeInDocuments`